packages/instrumenter/src/mutators/array-declaration-mutator.ts

```diff
@@ -1,25 +1,76 @@
 import babel, { type NodePath } from '@babel/core';
 
+// eslint-disable-next-line import/no-extraneous-dependencies
+import { Expression, SpreadElement, ArgumentPlaceholder, JSXNamespacedName } from '@babel/types';
+
 import { deepCloneNode } from '../util/index.js';
 
 import { NodeMutator } from './node-mutator.js';
 
 const { types } = babel;
 
+const arrayDeclarationReplacements = Object.assign({
+  EmptyArray: { replacementOperator: types.arrayExpression([types.stringLiteral('Stryker was here')]), mutatorName: 'EmptyArray' },
+  EmptyArrayConstructor: { replacementOperator: [types.stringLiteral('Stryker was here')], mutatorName: 'EmptyArrayConstructor' },
+  FilledArray: { replacementOperator: types.arrayExpression(), mutatorName: 'FilledArray' },
+  FilledArrayConstructor: { replacementOperator: [], mutatorName: 'FilledArrayConstructor' },
+} as const);
+
 export const arrayDeclarationMutator: NodeMutator = {
   name: 'ArrayDeclaration',
 
-  *mutate(path: NodePath): Iterable<babel.types.Node> {
-    if (path.isArrayExpression()) {
-      const replacement = path.node.elements.length ? types.arrayExpression() : types.arrayExpression([types.stringLiteral('Stryker was here')]);
-      yield replacement;
+  *mutate(path: NodePath, operations: string[] | undefined): Iterable<babel.types.Node> {
+    // The check of the [] construct in code
+    if (
+      path.isArrayExpression() &&
+      (operations == undefined ||
+        operations.includes(arrayDeclarationReplacements.EmptyArray.mutatorName as string) ||
+        operations.includes(arrayDeclarationReplacements.FilledArray.mutatorName as string))
```

Comment:
If you repeat the same if checks in the if body, do they make sense here? I think they can be removed but not 100% sure